crates/checkpoint-downloader/src/downloader.rs

```diff
@@ -265,7 +263,6 @@ impl ParallelCheckpointDownloaderInner {
         let mut last_scale = Instant::now();
         let mut consecutive_failures = 0;
         let average_workers = (downloader_config.min_workers + downloader_config.max_workers) / 2;
-        const MAX_CONSECUTIVE_POOL_MONITOR_FAILURES: u32 = 100;
```

Comment:
@sadhansood , how was this count originally determined? In PTN, it took nodes more than an hour to reach to 100 failures to check checkpoint lag.

I convert it into a config setting (and a few other parameters), so that we can easily change it.

Reply:
I did not want it to start adjusting workers immediately after full nodes go down and ~ 1 hour seems reasonable